pkg/envoy/config.go

```diff
@@ -1,9 +1,11 @@
 package envoy
 
-import k8s "github.com/openservicemesh/osm-health/pkg/kubernetes"
+import (
+	v1 "k8s.io/api/core/v1"
+)
 
-// GetEnvoyConfig returns the Envoy config for the given pod.
-func GetEnvoyConfig(pod k8s.Pod) (*Config, error) {
+// GetEnvoyConfigGetterForPod returns a ConfigGetter struct, which can fetch the Envoy config for the given pod.
+func GetEnvoyConfigGetterForPod(pod *v1.Pod) (ConfigGetter, error) {
 	// TODO(draychev): Get the config from the Envoy sidecar of the given pod.
 	return nil, nil
 }
```

Comment:
@johnsonshi Once you have the code to fetch Envoy config -- we'll reuse that function here to fetch config for a given pod.